### DISCUSSION TITLE
Not getting Source Files in Object Browser When library using a schema with a long schema name.

### DISCUSSION THREAD

**Discussion by SurroundLeePaul** [codefori/vscode-ibmi#2316] [Q&A] [answered]
I'm fairly new to VS Code for IBM i. Moving over from RDi.
I cannot see any of my source file objects in Libraries that have been converted to Long Schema names. For example, our LPFRTBIL Library is 
has a quoted Schema name of "LPT_FreightBill".

I narrowed it down to the CREATE TABLE command is using WHERE t.table_schema = 'LPFRTBIL'. I believe it should be using t.system_table_schema.

![image](https://github.com/user-attachments/assets/86a73db3-89c8-4940-a298-f97c292276a2)

**Comment by worksofliam** [comment]
@SurroundLeePaul Are you able to give us the steps to recreate this issue? Specifically how you did this part?

> converted to Long Schema names

**Reply by SurroundLeePaul** [reply]
@worksofliam, Thank you for your quick reply. Below is the SQL to create the Schema. If you then add a Source File to that Library, it will not show in the list.
```
CREATE SCHEMA "LPT_FreighBills" FOR LPFRTBIL;
CALL QSYS2.QCMDEXC ('CHGLIB LIB(LPFRTBIL) TYPE(*PROD) TEXT(''LPT Frieght Bills'')');
```

**Reply by worksofliam** [reply]
What happens if you run the select without the create? Does it return any results?

**Reply by SurroundLeePaul** [reply]
I'm not sure I understand the question. The create is how the library was created. We have many libraries created that way across several customers.

To clarify the issue:
Step 1: Create a Library with a long schema name similar to my example previously shown.
Step 2: Add a Source Physical File to the library.
Step 3: Open VS Code
Step 4: Connect to the IBM i where you created the Library and Source Physical File.
Step 5: Open the Object Browser
Step 5: add a filter for that Library
You will notice that it will not find the Source File you created in it.
This is happening, I believe, because the SQL used to get that list uses the table_schema field. to fix it, I suggest the SQL use system_table_schema.

**Reply by worksofliam** [reply]
I meant the select from your initial screenshot :D

**Reply by SurroundLeePaul** [reply]
that's not my SQL. IT is the SQL in the VS Code for IBM i solution.

**Reply by chrjorgensen** [reply]
@SurroundLeePaul I see your point - and you're right, we should use the system name columns instead of the SQL ones, since we display these values in the Object Browser tree. I'm working on a fix for this issue, stay tuned.

We've not encountered a situation like this before - I guess (almost) all users have libraries created by the CRTLIB command instead of SQL schemas (with long names).

**Reply by chrjorgensen** [reply, marked as answer]
@SurroundLeePaul We have just published a new release 2.13.6 containing a fix for your problem - please give it a spin and report back. 🙏

**Reply by SurroundLeePaul** [reply]
@chrjorgensen, Works great now. Thank you!
